clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy

```diff
@@ -112,28 +107,37 @@ class DeployOpenstackAtomicOperation implements AtomicOperation<DeploymentResult
       def stackName = serverGroupNameResolver.resolveNextServerGroupName(description.application, description.stack, description.freeFormDetails, false)
       task.updateStatus BASE_PHASE, "Heat stack name chosen to be ${stackName}."
 
-      task.updateStatus BASE_PHASE, "Loading templates"
-      String template = IOUtils.toString(this.class.classLoader.getResourceAsStream(TEMPLATE_FILE))
-      String subtemplate = IOUtils.toString(this.class.classLoader.getResourceAsStream(SUBTEMPLATE_FILE))
-      task.updateStatus BASE_PHASE, "Finished loading templates"
-
-      task.updateStatus BASE_PHASE, "Getting load balancer details for pool id $description.serverGroupParameters.poolId"
-      LbPool pool = provider.getLoadBalancerPool(description.region, description.serverGroupParameters.poolId)
-      task.updateStatus BASE_PHASE, "Found load balancer details for pool id $description.serverGroupParameters.poolId with name $pool.name"
-
-      task.updateStatus BASE_PHASE, "Getting internal port used for load balancer $pool.name"
-      int port = provider.getInternalLoadBalancerPort(pool)
-      task.updateStatus BASE_PHASE, "Found internal port $port used for load balancer $pool.name"
+      //look up all load balancer listeners -> pool ids and internal ports
```

Comment:
Can this be functionality be pulled out into a common method shared across upsertServerGroup/resizeServerGroup/upsertLoadBalancer?